cmd/auto-updater/main.go

```diff
@@ -213,6 +261,56 @@ func downloadArchive(ctx context.Context, file io.Writer, url string) (err error
 	return err
 }
 
+func unpackBinary(ctx context.Context, archive, target string) (err error) {
+	// TODO support different compression types e.g. tar.gz
+
+	zipReader, err := zip.OpenReader(archive)
+	if err != nil {
+		return err
+	}
+	defer func() { err = errs.Combine(err, zipReader.Close()) }()
+
+	if len(zipReader.File) != 1 {
+		return errors.New("archive contains more than one file")
+	}
+
+	zipedExec, err := zipReader.File[0].Open()
+	if err != nil {
+		return err
+	}
+	defer func() { err = errs.Combine(err, zipedExec.Close()) }()
+
+	newExec, err := os.OpenFile(target, os.O_CREATE|os.O_EXCL|os.O_WRONLY, os.FileMode(0755))
+	if err != nil {
+		return err
+	}
+	defer func() { err = errs.Combine(err, newExec.Close()) }()
+
+	_, err = sync2.Copy(ctx, newExec, zipedExec)
+	if err != nil {
+		return errs.Combine(err, os.Remove(newExec.Name()))
+	}
+	return nil
+}
+
+func restartSNService(name string) error {
+	switch runtime.GOOS {
+	case "windows":
+		// TODO how run this as one command `net stop servicename && net start servicename`?
+		_, err := exec.Command("net", "stop", name).Output()
```

Comment:
No idea about windows but I'm wondering if doesn't Windows have a restart command?

Reply:
at least I didn't find it